schematic/configuration.py

```diff
@@ -0,0 +1,98 @@
+import os
+import yaml
+
+
+class Configuration(object):
+
+    def __init__(self):
+        # this is the project's root directory
+        self.ROOT_DIR = None
+        # path to config.yml file
+        self.CONFIG_PATH = None
+        # path to credentials.json file
+        self.CREDS_PATH = None
+        # path to token.pickle file
+        self.TOKEN_PICKLE = None
+        # path to data directory
+        self.DATA_PATH = None
+        # path to service account credentials file
+        self.SERVICE_ACCT_CREDS = None
+        # path to synapse config file
+        self.SYNAPSE_CONFIG_PATH = None
+        # entire configuration data
+        self.DATA = None
+
+
+    def __getattribute__(self, name):
+        value = super().__getattribute__(name)
+        if value is None and "SCHEMATIC_CONFIG" in os.environ:
+            schematic_config = os.environ["SCHEMATIC_CONFIG"]
+            print(
+                "Loading config YAML file specified in 'SCHEMATIC_CONFIG' "
+                "environment variable: %s" % schematic_config
+            )
+            self.load_config(schematic_config)
+            value = super().__getattribute__(name)
+        elif value is None and "SCHEMATIC_CONFIG" not in os.environ:
+            raise AttributeError(
+                "The '%s' configuration field was accessed, but it hasn't been "
+                "set yet, presumably because the schematic.CONFIG.load_config() "
+                "method hasn't been run yet. Alternatively, you can re-run this "
+                "code with the 'SCHEMATIC_CONFIG' environment variable set to "
+                "the config.yml file, which will be automatically loaded."
+                % name
+            )
+        return value
+
+
+    def __getitem__(self, key):
+        return self.DATA[key]
+
+
+    @staticmethod
+    def load_yaml(file_path: str) -> dict:
```

Comment:
I moved the `load_yaml()` function to here because I would run into circular import errors otherwise, and it seemed like an inconsequential change since there was only that one function in the module where it came from.